docs/rfcs/0000-namespace.md

```diff
@@ -0,0 +1,115 @@
+- Start Date: (2015-10-29)
+- RFC PR: (leave this empty)
+- AMS Issue: (leave this empty)
+
+# Summary
+
+Provide a consistent API for the user of the AMS.
+
+# Motivation
+
+The actual public API is defined under `ActiveModelSerializers`,
+`ActiveModel::Serializer` and `ActiveModel`.
+
+At the `ActiveModel::Serializer` we have:
+
+- `ActiveModel::Serializer.config`
+- `ActiveModel::Serializer`
+
+At the `ActiveModelSerializers` we have:
+
+- `ActiveModelSerializers::Model`
+- `ActiveModelSerializers.logger`
+
+At `ActiveModel` we have:
+
+- `ActiveModel::SerializableResource`
+
+The idea here is to provide a single namespace `ActiveModel::Serializers` to the user.
```

Comment:
@maurogeorge thanks for all your work on this

We're going to need to either rewrite this doc or create a new one.  I'm going to enumerate below the result what's been happening in the gem and what we've been discussed in issues and in slack.

Scope of RFC: existing and future class and module names in gem used in namespacing. Does not include renaming of existing classes.

- We're not going to change the gem require lib. It remains `active_model_serializers`
- We're not going to change the namespace in the required file. It remains `ActiveModelSerializers`.
- We're going to wean ourselves off of `ActiveModel::Serializer` as the primary namespace.
- We're going to begin writing new code in `lib/active_model_serializers/` using the module namespace `ActiveModelSerializers`.
- We're going to move functionality, as we work in it, to live under  `lib/active_model_serializers/` and use the module namespace `ActiveModelSerializers`.
- For clarity, will we refer to the gem as `ActiveModelSerializers`, one word.
- Up for discussion: 
  - referring to it as `AMS`. Proposal: in discussion and code review yes, not in code or code comments or documentation, unless defined as an alias.
  - should `ActiveModel::SerializableResource` be moved under the `ActiveModelSerializers` namespace? Proposal: yes
  - should `ActiveModel::Serializer` be moved under the `ActiveModelSerializers` namespace? Proposal: yes
  - should `ActionController::Serialization` be moved under the `ActiveModelSerializers` namespace? Proposal: yes
  - should Adapters get their own namespace under `ActiveModelSerializers`? Proposal: yes
  - should Serializers get their own namespace under `ActiveModelSerializers`? Proposal: yes
  - should all `ActiveModel`-namespaced files in the gem be moved under `ActiveModelSerializers`? Proposal: yes
  - should moved classes or module will initially be aliased to their old name and location with deprecation warnings, such [as was done for `CollectionSerializer`](https://github.com/rails-api/active_model_serializers/pull/1251). Proposal: yes


Goal: 
  - clarify boundaries of gem and Rails
  - clarify that the `Serializer` class is no longer the primary behavior of the gem.

cc @beauby @NullVoxPopuli @joaomdmoura @kurko @steveklabnik